src/nvim/eval.c

```diff
@@ -15140,8 +15140,11 @@ static void f_sockconnect(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     return;
   }
 
+  snprintf((char *)IObuff, sizeof(IObuff), "%s:%" PRIdLINENR,
```

Comment:
Not sure how AOP comes in here. I just mean that stack information is natural to be "dynamically scoped" (that is, by global state saved/restored in C), rather than making it an argument for every function ever that could want to print a traceback, or just being part of such chain. This would make it simpler in the future to implement rich tracebacks (covering both vimL, lua and remote indirections) without needing to change ad-hoc logic spread around the codebase.